pkg/kvstore/etcd.go

```diff
@@ -505,38 +526,50 @@ func (e *etcdClient) Watch(w *Watcher) {
 					goto recreateWatcher
 				}
 
-				lastRev = r.Header.Revision
+				scopedLog := scopedLog.WithField(fieldRev, r.Header.Revision)
 
 				if err := r.Err(); err != nil {
-					log.WithFields(logrus.Fields{
-						fieldRev:     lastRev,
-						fieldWatcher: w,
-					}).WithError(err).Warningf("etcd watcher received error")
-					continue
+					// We tried to watch on a compacted
+					// revision that may no longer exist,
+					// recreate the watcher and try to
+					// watch on the next possible revision
+					if err == v3rpcErrors.ErrCompacted {
+						scopedLog.WithError(err).Debug("Tried watching on compacted revision")
```

Comment:
Maybe change to Info level, so that we may know how often this happens in production?
Or maybe increment a Prometheus metric?

Reply:
Agreed, filed https://github.com/cilium/cilium/issues/3022